packages/plugins/other/visitor-plugin-common/src/base-resolvers-visitor.ts

```diff
@@ -945,6 +940,92 @@ export class BaseResolversVisitor<
+          if (graphqlType instanceof GraphQLObjectType) {
+            const allInterfaces = graphqlType.getInterfaces();
+
+            if (allInterfaces.some(int => int.name === schemaType.name)) {
+              implementingTypes.push(graphqlType);
+            }
+          }
+        }
+
+        const possibleTypes =
+          implementingTypes
+            .map(type => {
+              // TODO: eddeee888 extract this and apply to ResolversUnionTypes
+              const isTypeMapped = this.config.mappers[type.name];
+              // 1. If mapped without placehoder, just use it without doing extra checks
+              if (isTypeMapped && !hasPlaceholder(isTypeMapped.type)) {
+                return { typename: type.name, typeValue: isTypeMapped.type };
+              }
+
+              // 2. Work out value for type
+              // 2a. By default, use the typescript type
+              let typeValue = this.convertName(type.name, {}, true);
+
+              // 2b. Find fields to Omit if needed.
+              //  - If no field to Omit, "type with maybe Omit" is typescript type i.e. no Omit
+              //  - If there are fields to Omit, keep track of these "type with maybe Omit" to replace in original unionMemberValue
+              const fieldsToOmit = this.getRelevantFieldsToOmit({
+                schemaType: type,
+                getTypeToUse: baseType => `RefType['${baseType}']`,
+              });
+              if (fieldsToOmit.length > 0) {
+                typeValue = this.replaceFieldsInType(typeValue, fieldsToOmit);
+              }
+
+              // 2c. If type is mapped with placeholder, use the "type with maybe Omit" as {T}
+              if (isTypeMapped && hasPlaceholder(isTypeMapped.type)) {
+                return { typename: type.name, typeValue: replacePlaceholder(isTypeMapped.type, typeValue) };
+              }
+
+              // 2d. If has default mapper with placeholder, use the "type with maybe Omit" as {T}
+              const hasDefaultMapper = !!this.config.defaultMapper?.type;
+              const isScalar = this.config.scalars[typeName];
+              if (hasDefaultMapper && hasPlaceholder(this.config.defaultMapper.type)) {
+                const finalTypename = isScalar ? this._getScalar(typeName) : typeValue;
+                return {
+                  typename: type.name,
+                  typeValue: replacePlaceholder(this.config.defaultMapper.type, finalTypename),
+                };
+              }
+              // ENDTODO
+              return { typename: type.name, typeValue };
+            })
+            .map(({ typename, typeValue }) => {
+              const nonOptionalTypenameModifier = this.config.resolversNonOptionalTypename.interfaceImplementingType
+                ? ` & { __typename: '${typename}' }`
+                : '';
+
+              return `( ${typeValue}${nonOptionalTypenameModifier} )`; // Must wrap every type in explicit "( )" to separate them
+            })
```

Comment:
This is the same logic as `ResolversUnionTypes`. 

I'll extract it out in a separate PR and after https://github.com/dotansimha/graphql-code-generator/pull/9231